### DISCUSSION TITLE
Using &RAMP Function to obtain ramped HRRPUA graph

### DISCUSSION THREAD

**Discussion by darwiethepang02** [firemodels/fds#12846] [Q&A]
Hello, I would like to ask if anyone can help me take a look at my code for my ramp function. I faced a problem where my ramp function stops about half of the assigned time as shown below in several cases: 

Case 1: I used HRRPUA: 9160 and simulate with RAMP function but the burning data stops at 420 seconds

&SURF ID = 'FIRE', HRRPUA = 9160., RAMP_Q='FRAMP', COLOR='ORANGE'/
&RAMP ID = 'FRAMP', T = 0, F = 0/
&RAMP ID = 'FRAMP', T = 200, F = 0/
&RAMP ID = 'FRAMP', T = 245.06, F = 0.03/
&RAMP ID = 'FRAMP', T = 284.58, F = 0.08/
&RAMP ID = 'FRAMP', T = 323.32, F = 0.12/
&RAMP ID = 'FRAMP', T = 363.64, F = 0.19/
&RAMP ID = 'FRAMP', T = 384.19, F = 0.27/
&RAMP ID = 'FRAMP', T = 393.68, F = 0.32/
&RAMP ID = 'FRAMP', T = 403.95, F = 0.60/
&RAMP ID = 'FRAMP', T = 415.81, F = 0.79/
&RAMP ID = 'FRAMP', T = 424.51, F = 0.85/
&RAMP ID = 'FRAMP', T = 440.32, F = 1.00/
&RAMP ID = 'FRAMP', T = 450.59, F = 0.91/
&RAMP ID = 'FRAMP', T = 455.34, F = 0.79/
&RAMP ID = 'FRAMP', T = 467.19, F = 0.65/
&RAMP ID = 'FRAMP', T = 482.21, F = 0.54/
&RAMP ID = 'FRAMP', T = 486.17, F = 0.50/
&RAMP ID = 'FRAMP', T = 496.44, F = 0.47/
&RAMP ID = 'FRAMP', T = 509.09, F = 0.49/
&RAMP ID = 'FRAMP', T = 525.69, F = 0.51/
&RAMP ID = 'FRAMP', T = 539.13, F = 0.54/
&RAMP ID = 'FRAMP', T = 546.25, F = 0.56/
&RAMP ID = 'FRAMP', T = 558.10, F = 0.57/
&RAMP ID = 'FRAMP', T = 566.01, F = 0.60/
&RAMP ID = 'FRAMP', T = 577.08, F = 0.70/
&RAMP ID = 'FRAMP', T = 587.35, F = 0.75/
&RAMP ID = 'FRAMP', T = 597.63, F = 0.87/
&RAMP ID = 'FRAMP', T = 615.81, F = 0.84/
&RAMP ID = 'FRAMP', T = 625.30, F = 0.89/
&RAMP ID = 'FRAMP', T = 637.94, F = 0.87/
&RAMP ID = 'FRAMP', T = 649.01, F = 0.81/
&RAMP ID = 'FRAMP', T = 660.08, F = 0.74/
&RAMP ID = 'FRAMP', T = 665.61, F = 0.77/
&RAMP ID = 'FRAMP', T = 675.89, F = 0.72/
&RAMP ID = 'FRAMP', T = 682.21, F = 0.67/
&RAMP ID = 'FRAMP', T = 700.40, F = 0.63/
&RAMP ID = 'FRAMP', T = 718.58, F = 0.57/
&RAMP ID = 'FRAMP', T = 727.27, F = 0.49/
&RAMP ID = 'FRAMP', T = 743.87, F = 0.42/
&RAMP ID = 'FRAMP', T = 764.43, F = 0.35/
&RAMP ID = 'FRAMP', T = 784.19, F = 0.31/
&RAMP ID = 'FRAMP', T = 804.74, F = 0.28/
&RAMP ID = 'FRAMP', T = 819.76, F = 0.27/
&RAMP ID = 'FRAMP', T = 831.62, F = 0.24/
&RAMP ID = 'FRAMP', T = 845.06, F = 0.21/
&RAMP ID = 'FRAMP', T = 864.82, F = 0.18/
&RAMP ID = 'FRAMP', T = 887.75, F = 0.17/
&RAMP ID = 'FRAMP', T = 901.19, F = 0.15/

Case 2: I used only the HRRPUA function but it still produced the same result in Case 1
&SURF ID = 'FIRE', HRRPUA = 9160./
 

Really appreciate it if someone can help me troubleshoot this. Thank you.

**Comment by drjfloyd** [comment]
Fuel must first mix with oxygen to burn, does your input provide for suffiicient oxgyen and for sufficient time for the fuel to mix with oxygen before unburned fuel leaves the domain?

**Reply by darwiethepang02** [reply]
I have placed a reaction on my fire source and the fuel source is ethanol. However, the burning is only occurring on the surface of the fire source as in it is the Z-axis is kept at one location only. Does this mean I need to increase the Z-axis of the reaction?

**Reply by drjfloyd** [reply]
No one knows exactly what your inputs are to be able to know what you may or may not need to do.  You also did not answer the question that was asked. You will need to provide an input file.

**Reply by darwiethepang02** [reply]
Sorry for the misunderstanding as I only started using FDS awhile ago and I am unsure what does it mean by having an input to provide for sufficient oxygen. Maybe can clarify on that? Would really appreciate. Thank you

**Reply by drjfloyd** [reply]
Fuel does not burn until it mixes with oxygen.  If you have insufficient oxygen (for example, you have consumed the oxygen intially present in the domain and did not provide for a suffiicent source of oxygen to continue burning) or if the fuel and oxygen do not have time to mix (for example, fuel leaves an OPEN VENT before it has time to mix with oxygen), then  that fuel will not burn.

**Reply by darwiethepang02** [reply]
I think I understand what you mean. So I will need to adjust my vents so that there will be sufficient oxygen available in the space?

**Reply by drjfloyd** [reply]
If the root cause is a lack of oxygen, then doing something to supply more oxygen (more OPEN vents for example) is needed.
If the root cause is a lack of time for fuel and oxygen to mix (e.g. the fire is near an OPEN boundary so fuel leaves before it can mx), then making the domain bigger can provide more time for mixing.

**Comment by obscureed** [comment]
Some (unofficial) comments to expand on drjfloyd's advice:
• At first sight, your ramp syntax looks correct. After 200s of delay, it sets up a two-peaked profile. (I wonder what else is happening during the first 200s -- if nothing, I would make the fire start earlier, or maybe use ``&TIME T_BEGIN=200 /``.)
• Your value 9160kW/m2 for HRRPUA is very intense. I'm not very familiar with ethanol pool fires, but I think it would need to be large to get into the thousands of kW/m2. You might find it more manageable to simulate a smaller example first.
• If you don't supply any OPEN vents at all, FDS will faithfully try to simulate a fire in an enclosed box. The pressure tends to get very high, and you might get a warning about weirdly high gas densities; eventually FDS might crash before the box runs out of oxygen. If you supply some OPEN vents (or other ways to get combustion products out and fresh air in), but not enough to ventilate the fire fully, FDS will try to simulate an underventilated fire. This is more difficult to simulate correctly than a fully ventilated fire. In general, you should set up OPEN vents where fresh air can enter and where combustion products can leave.
• To have a lack of oxygen, restricting combustion, does not necessarily mean zero oxygen. Check out the discussion of extinction models in the FDS User Guide.
• You said that combustion only happens at a thin surface in your model -- this is not the usual way that fires are set up in FDS. It is more likely that fuel is injected into the volumetric space of your model at that surface, and burns in a flame above it. The injected flowrate of fuel matches your specified HRRPUA (multiplied by the area of the fire surface), but (as drjfloyd has pointed out) fuel can escape from the model uncombusted.
• When you simulate a new application, it helps to have at least some postprocessing of many quantities: temperature, pressure, fuel mass fraction, oxygen mass fraction, heat release rate. (Maybe also search for "output quantity EXTINCTION" in the FDS User Guide.) For example, you could set up a vertical slice through the fire bed with each of those quantities. Check that all of the results are reasonable.
• An interesting question might be "how much combustion is happening in the first Z metres of the flame?". To answer this, you can set up monitors such as:
```
&DEVC XB=  [[xmin,xmax,  ymin,ymax]]  0.0, 1.0, QUANTITY='HRRPUV', STATISTICS='VOLUME INTEGRAL',  ID='hrr_z_0_to_1.0' /
&DEVC XB=  [[xmin,xmax,  ymin,ymax]]  0.0, 2.0, QUANTITY='HRRPUV', STATISTICS='VOLUME INTEGRAL',  ID='hrr_z_0_to_2.0' /
```
etc (where you put in values instead of `[[xmin etc]]` to include all of the flame, maybe all of the model). If the values are still rising when the monitors reach the top of your model, you are missing some of the flame.
